bundles/org.eclipse.ui.ide.application/src/org/eclipse/ui/internal/ide/application/DelayedEventsProcessor.java

```diff
@@ -183,90 +198,119 @@ public void run() {
 					String msg = NLS.bind(IDEWorkbenchMessages.OpenDelayedFileAction_message_fileNotFound, initialPath);
 					MessageDialog.open(MessageDialog.ERROR, window.getShell(),
 							IDEWorkbenchMessages.OpenDelayedFileAction_title, msg, SWT.SHEET);
-				} else if (details.fileInfo.isDirectory()) {
-					SmartImportWizard wizard = new SmartImportWizard();
-					wizard.setInitialImportSource(new File(details.fileStore.toURI()));
-					WizardDialog dialog = new WizardDialog(window.getShell(), wizard);
-					dialog.setBlockOnOpen(false);
-					dialog.open();
-				} else {
-					IWorkbenchPage page = window.getActivePage();
-					if (page == null) {
-						String msg = NLS.bind(IDEWorkbenchMessages.OpenDelayedFileAction_message_noWindow,
-								details.path);
-						MessageDialog.open(MessageDialog.ERROR, window.getShell(),
-								IDEWorkbenchMessages.OpenDelayedFileAction_title, msg, SWT.SHEET);
-					}
-					try {
-						IEditorPart openEditor = IDE.openInternalEditorOnFileStore(page, details.fileStore);
-						Shell shell = window.getShell();
-						if (shell != null) {
-							if (shell.getMinimized())
-								shell.setMinimized(false);
-							shell.forceActive();
-						}
-
-						if (details.line >= 1) {
-							try {
+					return;
+				}
+				IWorkbenchPage page = window.getActivePage();
+				if (page == null) {
+					String msg = NLS.bind(IDEWorkbenchMessages.OpenDelayedFileAction_message_noWindow,
+							details.path);
+					MessageDialog.open(MessageDialog.ERROR, window.getShell(),
+							IDEWorkbenchMessages.OpenDelayedFileAction_title, msg, SWT.SHEET);
+					return;
+				}
+				Shell shell = window.getShell();
+				if (shell != null) {
+					if (shell.getMinimized())
+						shell.setMinimized(false);
+					shell.forceActive();
+				}
+				try {
+					if (details.fileInfo.isDirectory()) {
+						// Open a directory
+						File directory = new File(details.fileStore.toURI());
+						for (IProject project : ResourcesPlugin.getWorkspace().getRoot().getProjects()) {
+							IPath location = project.getLocation();
+							if (location != null && directory.equals(location.toFile())) {
 								/*
-								 * Do things with reflection to avoid having to rely on the text editor
-								 * plug-ins.
+								 * The directory is the root of an existing project.
+								 * Select in Project Explorer.
+								 * If open, then refresh, else open.
 								 */
-								Bundle textEditorBundle = Platform.getBundle(TEXTEDITOR_BUNDLE_NAME);
-								if (textEditorBundle != null) {
-									Class<?> textEditorClass = textEditorBundle.loadClass(TEXTEDITOR_CLASS_NAME);
-									if (textEditorClass != null) {
-										Object textEditor = invoke(openEditor, "getAdapter", //$NON-NLS-1$
-												new Class[] { Class.class }, new Object[] { textEditorClass });
-										if (textEditor != null) {
-											openEditor = (IEditorPart) textEditor;
-										}
+								IViewPart view = page.showView(ProjectExplorer.VIEW_ID);
```

Comment:
So this is specific to the project explorer view. But what about other views like "Package Explorer" or other "Navigator-Views"?

This will open the project explorer view if it is not yet open, correct? Is this really what a user wants?

Reply:
I think this is OK, considering that "old" Navigator is not there, Package Explorer may not be installed (CDT etc), Git repo Explorer is Git specific.

Reply:
But do we really need this. What would happen if we don't show and refresh that view. Wouldn't the editor do that refresh implicitly when the file is shown in the editor?

Reply:
I would expect it would be surprising if someone wants to see a directory (from command line) but IDE would "silently" do that in background and no visual feedback is given. But may be I misunderstood the request here. If I understand it right, one passes directory path to IDE and IDE is supposed to show appropriate resource in a view. For that I would consider showing the (potentially hidden / closed) view as "must have" part of the enhancement.

Reply:
> I would expect it would be surprising if someone wants to see a directory (from command line) but IDE would "silently" do that in background and no visual feedback is given. But may be I misunderstood the request here. If I understand it right, one passes directory path to IDE and IDE is supposed to show appropriate resource in a view. For that I would consider showing the (potentially hidden / closed) view as "must have" part of the enhancement.

You are right. For a directory this makes sense - as no editor opens. But what if we open a file that is inside an project in the workspace? I think in that case we should not "touch" the view but leave it to the editor to "refresh" the view.

Reply:
OK, that's correct, view should only be opened for directories.

Reply:
But we are inside `if (details.fileInfo.isDirectory()` so that is already given